### ISSUE TITLE
[kubeapps-apis] Implement packaging related functionality for fluxv2 plugin

### ISSUE THREAD

**Issue by absoludity** [kubeapps/kubeapps#2946]
Once #2944 is ready, we want to implement the packaging API (only, ie. not yet api functionality for repositories) for the fluxv2 plugin (as well as one other plugin, but perhaps that should be a separate issue - see #2993 for the direct helm plugin).

This will involve something similar to (subject to name changes):

- [x] GetAvailablePackagesSummaries
- [x] GetAvailablePackageDetail
- [x] GetAvailablePackageVersions
- [x] GetInstalledPackageSummaries
- [x] GetInstalledPackageDetail
- [x] CreateInstalledPackage
- [x] UpdateInstalledPackage
- [ ] DeleteInstalledPackage

**Comment by absoludity** [comment]
I think we've agreed here that we'll just do the first 3 initially. Also, as part of this Greg will be looking at providing a caching layer initially specifically to the flux plugin, but generalising it to provide caching to plugins. It may be worth splitting that to a separate issue.

**Comment by gfichtenholt** [comment]
yes, that's correct, that was the agreement. As far as splitting into separate issue  is concerned, doesn't matter to me, if you think that's best, go ahead or if you prefer let me know and I will open one

**Comment by absoludity** [comment]
IMO it's a matter of clearly communicating what we're currently working on. If you think that the cache investigation and work is going to be significant in its own right (as in, something to report progress on at the end of the iteration), then yep, I'd recommend splitting it out, briefly outlining what you expect to be involved, so you can updated it to make it easier for Pepe at the end of the iteration to summarise what's been happening in the world of Kubeapps. You could also just do that here if you want, but the board (which some people use to see what we're doing) won't reflect that of course.

**Comment by gfichtenholt** [comment]
created https://github.com/kubeapps/kubeapps/issues/3032

**Comment by antgamdia** [comment]
I'd have a question (perhaps it was already said, but I don't remember now): is the Fluxv2 plugin going to support rolling back releases (even if it is not natively supported by flux)? Or, given that Flux does not support it, we won't support it?

**Comment by gfichtenholt** [comment]
yes, that should be doable. If I remember correctly, manual rollback of a release was an example given to me by flux authors in a slack discussion in that one has do invoke the underlying helm client to do it.
https://cloud-native.slack.com/archives/CLAJ40HV3/p1629245848190300?thread_ts=1629244025.187100&cid=CLAJ40HV3

**Comment by antgamdia** [comment]
No, no, I don't think we need it soon; it is just something I just was curious about. It is related to what I mention in my PR https://github.com/kubeapps/kubeapps/pull/3479#issuecomment-927106955
Thanks for the info and context!
